wiki-webui/src/main/java/org/exoplatform/wiki/commons/Utils.java

```
@@ -186,30 +193,42 @@ public static String getURLFromParams(WikiPageParams params) throws Exception {
+      spaceUrl.append(getWikiUriOfCurrentSpace()).append("/");
       if (!StringUtils.isEmpty(params.getPageName())) {
         spaceUrl.append(params.getPageName());
       }
       return spaceUrl.toString();
     }
     return org.exoplatform.wiki.utils.Utils.getPermanlink(params, false);
   }
-  
-  private static String getWikiAppNameInSpace(String spaceId) {
-    SpaceService spaceService = org.exoplatform.wiki.rendering.util.Utils.getService(SpaceService.class);
-    Space space = spaceService.getSpaceByGroupId(spaceId);
-    String apps = space.getApp();
-    if (apps != null) {
-      for (String app : apps.split(",")) {
-        String[] appInfos = app.split(":");
-        if (appInfos.length > 1 && "WikiPortlet".equals(appInfos[0])) {
-          return appInfos[1];
+
+  private static String getWikiUriOfCurrentSpace() {
+    SpaceService spaceService = (SpaceService) PortalContainer.getComponent(SpaceService.class);
+    try {
+      String currentWikiNodeUri = getCurrentWikiNodeUri();
+      String currentSpaceName = getCurrentSpaceName();
+      if (currentWikiNodeUri.contains(":spaces:")) {
+        // "/portal/g/:spaces:testSpace/testSspace/wiki_uri/testPage" pattern usecase
+        String[] parsedUrl = currentWikiNodeUri.split("/");
+        if (ArrayUtils.contains(parsedUrl, currentSpaceName)) {
+          int index = ArrayUtils.indexOf(parsedUrl, currentSpaceName);
+          return parsedUrl[index + 1];
+        }
+      } else {
+        // "/portal/intranet/wiki/group/spaces/testSpace/testPage" pattern usecase
+        Collection<UserNode> children =  SpaceUtils.getSpaceUserNode(spaceService.getSpaceByUrl(currentSpaceName)).getChildren();
+        for(UserNode child : children) {
+          if (child.getName().contains("wiki")){
+            return child.getName();
+          }
         }
       }
+    } catch (Exception e) {
+      LOG.warn("Get Wiki Uri Of Current Space failed.");

```

Do not swallow the exception.